ci(workflows): Add Tar Zipped Asset for Helm Chart Sample Consumption

eshenayo opened the pull request:
<!-- Expected Commit Message Description (imported automatically by GitHub) -->
<!-- Must conform to [conventional commits guidelines](https://github.com/edgexfoundry/edgex-examples/blob/main/.github/Contributing.md) -->
<!-- Expected Commit message must contain Closes/Fixes #IssueNumber statement when there is a related issue -->

This provides the ability to consume the helm chart example as an asset. It is for use with automation in another system that requires this file format for consumption and installation of the chart.

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) - I did not due to time constraints.
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)

## Testing Instructions
To see the GitHub Action in..."action" (I'll see myself out), create a release

* Successful run of the action can be found here: https://github.com/eshenayo/edgex-examples/runs/5862759176?check_suite_focus=true
* Creates a tar zipped file containing the Helm deployment directory: https://github.com/eshenayo/edgex-examples/releases/tag/v0.1.0-alpha
![image](https://user-images.githubusercontent.com/8902109/171455968-9a1aa256-a9a8-4dd3-8e03-1dffacc3821b.png)

bnevis-i reviewed (changes requested):
This patch is corrupt (67 files changed) and needs to be rebased off of upstream/main.

eshenayo commented:
> > This patch is corrupt (67 files changed) and needs to be rebased off of upstream/main.
> 
> Thanks for that catch, I didn't see that the number of files changed after I modified the sign-off, I'll fix that.

@bnevis-i Fixed commit/merge, now only 1 file shows modified. 
![image](https://user-images.githubusercontent.com/8902109/171907793-4aca2c04-db1b-4e63-ad7a-7c58d7138abf.png)

Let me know if there are any further changes needed.

lenny-intel commented:
@JamesKButcher , please add this to the agenda for this week's Core WG. THX!

bnevis-i reviewed (approved):
Approving to remove disapproving review.  Have no problem with this.  Core WG will decide if we WANT to do this.
